tests/test_scheduler.py

```diff
@@ -1,47 +1,115 @@
+    #SBATCH --account={firecrest_config.builder_metadata_options_account}
+    #SBATCH --nodes=1
+    #SBATCH --ntasks-per-node=1
+    #SBATCH --constraint=mc
+    #SBATCH --mem=10K
 
-    _script = Path(tmp_path / "job.sh")
-    _script.write_text("#!/bin/bash\n\necho 'hello world'")
+    echo 'hello world'
+    """
+
+    dedented_script = textwrap.dedent(shell_script).strip()
+    Path(tmpdir / "job.sh").write_text(dedented_script)
+    remote_ = transport._cwd.joinpath(firecrest_config.workdir, "job.sh")
+    transport.put(tmpdir / "job.sh", remote_)
+
+    job_id = scheduler.submit_job(firecrest_config.workdir, "job.sh")
 
-    job_id = scheduler.submit_job(transport.getcwd(), _script)
-    # this is how aiida expects the job_id to be returned
     assert isinstance(job_id, str)
 
 
+@pytest.mark.timeout(180)
 @pytest.mark.usefixtures("aiida_profile_clean")
-def test_get_jobs(firecrest_computer: orm.Computer):
+def test_get_and_kill_jobs(
+    firecrest_computer: orm.Computer, firecrest_config, tmpdir: Path
+):
+    """Test getting and killing jobs from the scheduler.
+    We test the two together for performance reasons, as this test might run against
+      a real server and we don't want to leave parasitic jobs behind.
+      also less billing for the user.
+    Note: this test relies on a functional transport.put() method.
+    """
     transport = firecrest_computer.get_transport()
     scheduler = FirecrestScheduler()
     scheduler.set_transport(transport)
 
-    # test pagaination
-    scheduler._DEFAULT_PAGE_SIZE = 2
-    Values._DEFAULT_PAGE_SIZE = 2
+    # don't raise in case of invalid job id
+    scheduler.get_jobs(["000"])
 
-    joblist = ["111", "222", "333", "444", "555"]
+    shell_script = f"""
+    #!/bin/bash
+    #SBATCH --no-requeue
+    #SBATCH --job-name="aiida-1929"
+    #SBATCH --get-user-env
+    #SBATCH --output=_scheduler-stdout.txt
+    #SBATCH --error=_scheduler-stderr.txt
+    #SBATCH --account={firecrest_config.builder_metadata_options_account}
+    #SBATCH --nodes=1
+    #SBATCH --ntasks-per-node=1
+    #SBATCH --constraint=mc
+    #SBATCH --mem=10K
+
+    sleep 180
+    """
+
+    joblist = []
+    dedented_script = textwrap.dedent(shell_script).strip()
+    Path(tmpdir / "job.sh").write_text(dedented_script)
+    remote_ = transport._cwd.joinpath(firecrest_config.workdir, "job.sh")
+    transport.put(tmpdir / "job.sh", remote_)
+
+    for _ in range(5):
+        joblist.append(scheduler.submit_job(firecrest_config.workdir, "job.sh"))
+
+    # test pagaination is working
+    scheduler._DEFAULT_PAGE_SIZE = 2
     result = scheduler.get_jobs(joblist)
     assert len(result) == 5
     for i in range(5):
-        assert result[i].job_id == str(joblist[i])
+        assert result[i].job_id in joblist
         # TODO: one could check states as well
 
+    # test kill jobs
+    for jobid in joblist:
+        scheduler.kill_job(jobid)
+
+    # sometimes it takes time for the server to respond
+    sleep(5)
```

Comment:
Can't you do a while loop checking the results every 1e-3 or 1e-2 seconds with a timeout of 5 seconds? That makes the tests faster. I guess this is a pattern that will be more frequently used in the tests.

Reply:
good idea, I put 0.5 second, as sending the request every time takes about similar time, so we don't want to send it too often.